File: merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        l=len(intervals)
        arr=[]
        intervals.sort()
        arr.append(intervals[0])
        print(arr[-1][1])
        for i in range(1, len(intervals)): 
            
            # There is intersection, then update the last sub-list of result
            if intervals[i][0] <= arr[-1][1]:  
                arr[-1][1] = max(arr[-1][1], intervals[i][1])
                
                
            else:
                arr.append(intervals[i])
        return arr
    

File: merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        List<int[]> ls = new ArrayList<int[]>();
        Arrays.sort(intervals,(i1,i2)->Integer.compare(i1[0], i2[0]));
        ls.add(intervals[0]);
        for(int i = 0; i < intervals.length; i++) {
            if(ls.get(ls.size()-1)[1]>=intervals[i][0]){
                 ls.get(ls.size() - 1)[1] = Math.max(ls.get(ls.size() - 1)[1], intervals[i][1]);
            }
            else{
                ls.add(intervals[i]);
            }
        }
        return ls.toArray(new int[ls.size()][]);

    }
}